sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -84,6 +84,22 @@ int pollSocketUntilEventOrTimeout(
   return WSAPoll(&poller, 1, timeout);
 #endif
 }
+
+#ifdef WINDOWS
```

Comment:
Is there no other way to solve the issue without having to special case the code for Windows vs Linux?

Reply:
this is how libcurl does it when it is used with the "proper" way.
since we are getting closer to the platform working with sockets, it is common to find this types of differences across OS.